CONTRIBUTING.md

```diff
@@ -1,42 +1,231 @@
+          |    | Manages test execution. |
+          |    '-------------------------'
+          |
+          |    .----------------------------------------.
+          |    |                   io                   |
+          |--->|----------------------------------------|
+          |    | Manages UI, logging, and test records. |
+          |    '----------------------------------------'
+          |
+          |    .---------------------------------------------------.
+          |    |                       plugs                       |
+          |--->|---------------------------------------------------|
+          |    | Extensions for interfacing with various hardware. |
+          |    '---------------------------------------------------'
+          |
+          |    .-----------------------------------.
+          |    |               util                |
+          '--->|-----------------------------------|
+               | Utility functions and misc tools. |
+               '-----------------------------------'
+```
+
+
+## Setting Up Your Dev Environment
 The OpenHTF codebase is set up to use pip/virtualenv to manage dependencies.
 
 Pip is the python package manager, while virtualenv is a tool to isolate python
 environments. You'll need both in order to work with the OpenHTF codebase.
 
-Pip installs requirements (dependencies) from pypi (the python package index).
-It can use a requirements.txt file to know which packages to install. Typically
-this file points at hosted package names in pypi to install but it can also
-install local packages in development mode (using the -e flag).  This causes pip
+Pip installs dependencies from pypi (the python package index), but it can also
+install local packages in development mode (using the -e flag). This causes pip
 to install the dependency using symlinks from the filesystem so changes made to
 those files are instantly reflected.
 
-By design, virtualenv and OpenHTF's python dependencies are not included in the
+Neither virtualenv nor OpenHTF's python dependencies are included in the
 repository. The following steps set up a new virtualenv environment and install
-OpenHTF's dependencies into it using pip (steps 5-7 below).
+OpenHTF's dependencies into it using pip.
+
+
+### Linux
+```bash
+# Clone into the repo.
+git clone git@github.com:google/openhtf.git
 
+# Install system-level third-party dependencies.
+sudo apt-get install python-pip swig libssl-dev python-dev libffi-dev \ 
+protobuf-compiler
 
-## Dev Environment Setup Steps
-To set up an OpenHTF dev environment, follow the steps below.
+# Make sure pip is up-to-date.
+sudo pip install --upgrade pip
+
+# Install virtualenv via pip.
+sudo pip install virtualenv
+
+# Change to the openhtf directory.
+cd openhtf
+
+# Create a new virtualenv.
+virtualenv venv
+
+# Activate the new virtualenv.
+. venv/bin/activate
+
+# Install openhtf into the virtualenv in dev mode.
+python setup.py develop
+```
+
+
+## Web Frontend Development
```

Comment:
+1, i would just remove any delta to the .md files from the other PR to avoid any merge conflicts, we'll get this one in ASAP anyway so if they don't match for one commit, that's fine.